lib/src/trace.dart

```diff
@@ -103,38 +109,41 @@ class Trace implements StackTrace {
 
     if (trace is Trace) return trace;
     if (trace is Chain) return trace.toTrace();
-    return new LazyTrace(() => new Trace.parse(trace.toString()));
+    return new LazyTrace(() => new Trace.parse(trace.toString(), trace));
   }
 
   /// Parses a string representation of a stack trace.
   ///
   /// [trace] should be formatted in the same way as a Dart VM or browser stack
   /// trace. If it's formatted as a stack chain, this will return the equivalent
   /// of [Chain.toTrace].
-  factory Trace.parse(String trace) {
+  factory Trace.parse(String trace, [StackTrace originalTrace]) {
```

Comment:
`originalTrace` should be a named parameter. Positional parameters are generally not great for public APIs, since it makes it impossible to add named parameters later on.